src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+
+    data.forEach((item) => {
+      renderCard(item);
+    });
+  })
+
+  .catch((err) => {
+    console.error(err);
+  });
+
+//functions
+function renderCard(item, method = "prepend") {
+  const cardElement = getCardElement(item);
+  cardsList[method](cardElement);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+  modal.removeEventListener("mousedown", handleOutsideClick);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameElement = cardElement.querySelector(".card__text");
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameElement.textContent = data.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardDeleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardLikeButton.addEventListener("click", (evt) => {
+    handleLike(evt, data._id);
+  });
+
+  function handleLike(evt, id) {
+    let isLiked = evt.target.classList.contains("card__like-button_liked");
+
+    api
+      .handleLikeStatus(id, isLiked)
+      .then(() => {
+        evt.target.classList.toggle("card__like-button_liked");
+      })
+      .catch(console.error);
+  }
+
+  cardImage.addEventListener("click", () => {
+    openModal(cardPreview);
+    cardPreviewImage.src = data.link;
+    cardPreviewImage.alt = data.name;
+    cardPreviewCaption.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+  modal.addEventListener("mousedown", handleOutsideClick);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Delete");
+    });
+}
+
+closeButtons.forEach((button) => {
+  const popup = button.closest(".modal");
+
+  button.addEventListener("click", () => closeModal(popup));
+});
+
+profileEditButton.addEventListener("click", () => {
+  profileInputName.value = profileName.textContent;
+  profileInputDescription.value = profileDescription.textContent;
+  resetValidation(
+    profileForm,
+    [profileInputName, profileInputDescription],
+    settings
+  );
+  openModal(editModal);
+
+  deleteModalCancelBtn.addEventListener("click", () => {
+    closeModal(deleteModal);
+  });
+});
```

Comment:
* The `cancel` button doesn't work https://snipboard.io/BQw6G7.jpg 


It's because the listener is **inside** another listener, but it should be just in the file body.


Please, add it **outside** any handler or function